src/systems/systems_wasm/src/aspects.rs

```diff
@@ -0,0 +1,852 @@
+        self.variables.write(&identifier, value);
+    }
+
+    /// Copy a variable's value to another variable.
+    pub fn copy(&mut self, input: Variable, output: Variable) {
+        Self::precondition_not_aircraft_variable(&output);
+
+        let input = self.variables.register(&input);
+        let output = self.variables.register(&output);
+
+        self.actions.push((
+            Map::new(input, |value| value, output).into(),
+            ExecuteOn::PreTick,
+        ));
+    }
+
+    /// Map a variable's value to another variable, applying the given function in the process.
+    pub fn map(
+        &mut self,
+        execute_on: ExecuteOn,
+        input: Variable,
+        func: fn(f64) -> f64,
+        output: Variable,
+    ) {
+        Self::precondition_not_aircraft_variable(&output);
+
+        let inputs = self.variables.register(&input);
+        let output = self.variables.register(&output);
+
+        self.actions
+            .push((Map::new(inputs, func, output).into(), execute_on));
+    }
+
+    /// Map a set of variable values to another variable.
+    pub fn map_many(
+        &mut self,
+        execute_on: ExecuteOn,
+        inputs: Vec<Variable>,
+        func: fn(&[f64]) -> f64,
+        output: Variable,
+    ) {
+        Self::precondition_not_aircraft_variable(&output);
+
+        let inputs = self.variables.register_many(&inputs);
+        let output = self.variables.register(&output);
+
+        self.actions
+            .push((MapMany::new(inputs, func, output).into(), execute_on));
+    }
+
+    /// Reduce a set of variable values into one output value and write it to a variable.
+    pub fn reduce(
+        &mut self,
+        execute_on: ExecuteOn,
+        inputs: Vec<Variable>,
+        init: f64,
+        func: fn(f64, f64) -> f64,
+        output: Variable,
+    ) {
+        Self::precondition_not_aircraft_variable(&output);
+
+        let inputs = self.variables.register_many(&inputs);
+        let output = self.variables.register(&output);
+
+        self.actions
+            .push((Reduce::new(inputs, init, func, output).into(), execute_on));
+    }
+
+    /// Write a set of variables to an object.
+    pub fn variables_to_object(&mut self, instance: Box<dyn VariablesToObject>) {
+        let variables = self.variables.register_many(&instance.variables());
+
+        self.actions.push((
+            ToObject::new(instance, variables).into(),
+            ExecuteOn::PostTick,
+        ));
+    }
+
+    /// Convert event occurrences to a variable.
+    ///
+    /// If you want to write a variable back to the same event, then use the
+    /// event id returned by this method and pass it to [Self::variable_to_event_id].
+    pub fn event_to_variable(
+        &mut self,
+        event_name: &str,
+        mapping: EventToVariableMapping,
+        target: Variable,
+        configure_options: fn(EventToVariableOptions) -> EventToVariableOptions,
```

Comment:
Why did you opt for making this a function from `EventToVariableOptions` to the same type? I think just using `option: EventToVariableOptions` would reduce complexity, rather than allowing a `configure_options` function and always passing in the default value.

Reply:
This lets one write:

```rust
    builder.event_to_variable(
        "BRAKES",
        EventToVariableMapping::SmoothPress(KEYBOARD_PRESS_SPEED, KEYBOARD_RELEASE_SPEED),
        Variable::aspect("BRAKES"),
        |options| options.mask(),
    )?;
```

Whereas changing it to `EventToVariableOptions` would require the following code:

```rust
    builder.event_to_variable(
        "BRAKES",
        EventToVariableMapping::SmoothPress(KEYBOARD_PRESS_SPEED, KEYBOARD_RELEASE_SPEED),
        Variable::aspect("BRAKES"),
        EventToVariableOptions::default().mask(),
    )?;
```

I like having the type not repeated on every usage.